cmd/analyzetestresults/analyzetestresults.go

```diff
@@ -0,0 +1,221 @@
+						tcMessage = returnContentWrappedInDropdown(dropdownSummaryString, tc.SystemErr)
+					case tc.Failure != nil:
+						tcMessage = "```\n" + tc.Failure.Message + "\n```"
+					default:
+						tcMessage = "```\n" + tc.Error.Message + "\n```"
+					}
+
+					testCaseEntry := "* :arrow_right: " + "[**`" + tc.Status + "`**] " + tc.Name + "\n" + tcMessage
+					f.failedTestCaseNames = append(f.failedTestCaseNames, testCaseEntry)
+				}
+			}
+		}
+	}
+}
+
+// getFormattedReport returns the full report as a string
+func (f *FailedTestCasesReport) getFormattedReport() string {
+	msg := f.headerString
+	for _, failedTCName := range f.failedTestCaseNames {
+		msg += fmt.Sprintf("\n %s\n", failedTCName)
+	}
+
+	return msg
+}
+
+func returnContentWrappedInDropdown(summary, content string) string {
+	return "<details><summary>" + summary + "</summary><br><pre>" + content + "</pre></details>"
+}
+
+// AnalyzeTestResultsCmd represents the analyze-test-results command
+var AnalyzeTestResultsCmd = &cobra.Command{
+	Use:   "analyze-test-results",
+	Short: "Command for analyzing test results",
+	PreRunE: func(cmd *cobra.Command, _ []string) error {
+		if viper.GetString(types.OciArtifactRefParamName) == "" {
+			_ = cmd.Usage()
+			return fmt.Errorf("parameter %q not provided, neither %s env var was set", types.OciArtifactRefParamName, types.OciArtifactRefEnv)
+		}
+		return nil
+	},
+	SilenceUsage: true,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		ociArtifactRef = viper.GetString(types.OciArtifactRefParamName)
+
+		cfg := oci.ScannerConfig{
+			OciArtifactReference: ociArtifactRef,
+			FileNameFilter:       []string{types.JunitFilename, clusterProvisionLogFilename, e2eTestRunLogFilename},
+		}
+
+		scanner, err := oci.NewArtifactScanner(cfg)
+		if err != nil {
+			return fmt.Errorf("failed to initialize artifact scanner: %+v", err)
+		}
+
+		if err := scanner.Run(); err != nil {
+			return fmt.Errorf("failed to scan artifact from %s: %+v", ociArtifactRef, err)
+		}
+
+		overallJUnitSuites, err := getTestSuitesFromXMLFile(scanner, jUnitFilename)
+		// make sure that the Prow job didn't fail while creating the cluster
+		if err != nil && !strings.Contains(err.Error(), fmt.Sprintf("couldn't find the %s file", jUnitFilename)) {
+			return fmt.Errorf("failed to get JUnitTestSuites from the file %s: %+v", jUnitFilename, err)
+		}
+
+		failedTCReport := FailedTestCasesReport{}
+
+		for _, file := range scanner.FilesPathMap {
+			if file.Filename == clusterProvisionLogFilename {
+				failedTCReport.clusterProvisionLog = file.Content
+			}
+			if file.Filename == e2eTestRunLogFilename {
+				failedTCReport.e2eTestLog = file.Content
+			}
+		}
+
+		failedTCReport.setHeaderString(overallJUnitSuites)
+		failedTCReport.extractFailedTestCases(overallJUnitSuites)
+
+		if err := os.WriteFile(outputFilename, []byte(failedTCReport.getFormattedReport()), 0o600); err != nil {
+			return fmt.Errorf("failed to create a file with the test result analysis: %+v", err)
```

Comment:
I think rather than passing AllJunitSuites, we pass these functions the Report struct. Rather than these functions being apart of the Report struct, they are apart of this cmd struct and more for the purpose of formatting than failure detection and formatting.

Reply:
Hmm I think it'd be good to expand the testresults package on functionality of generating the test report in a human readable format and just let this cmd to request the formatted report. But I'm open to a discussion